### DISCUSSION TITLE
2 WiFi-Cards and 2 Inverters

### DISCUSSION THREAD

**Discussion by Qerox3** [wlcrs/huawei_solar#777] [Setting up the integration]
Hello, im running 1 10KTL and 1 4KTL, both are installed by a company so i can't change or even see any modbus settings.

Connecting to the 10KTL Wifi with Modbus 0 gets me the 10KTL Data, Battery Data and Smart-Meter Data
Connecting to the 4KTL Wifi with Modbus 0 gets me the 4KTL Data
Connecting to the SDongle with Modbus 1 gets me the 4 KTL Data

Trying out ID 0,X (other IDs) didn't work.

So the only idea left was to add another WiFi-Card (cheap Wifi USB Stick) to the Pi and let the internal wifi connect to inverter 1 and the wifi stick to inverter 2. But, seems like because of the IP configuration which makes sense because the wifi-card being the client and the inverters being the router, it also doesn't work.

Maybe someone more HASS/Linux/Huawei/Network experienced knows a workaround for this issue?

Those are the Auto assigned IPs, switching to a static gateway IP doesnt work.
![PHOTO-2024-07-24-16-37-53 2](https://github.com/user-attachments/assets/986525f6-dca9-4cb8-80f3-816beaeb1f59)
![PHOTO-2024-07-24-16-37-53](https://github.com/user-attachments/assets/fe1cdcd8-de85-47e4-9f62-3233f06e1cd3)

**Comment by achtbitunix** [comment]
Hi,

two docker container could be a solution.
A good start to do that is maybe this: https://github.com/fgg89/docker-ap/wiki/Container-access-to-wireless-network-interface

Or using network namespaces like this one: https://blogs.igalia.com/dpino/2016/04/10/network-namespaces/

**Comment by Kugelfang666** [comment]
I have a similar solution: Easy Fix change the IP range for one of the inverters

I have two inverters

Inverter 1: 192.168.20**0**.XXX
Inverter 2: 192.168.20**1**.XXX

My Home Assistant has two USB WiFi cards, each joined to the AP of one Inverter. 

this way was able to load this integration twice with two different IPs

**Reply by annoyedineedausername** [reply]
How do you change the IP range of the inverters?

**Reply by Kugelfang666** [reply]
@annoyedineedausername if you log into the inverter with the local AP using the Fusion Solar App you can configure its DHCP

**Reply by codex70** [reply]
Don't you need an installer account to log in to the access point? I can connect with my phone to the AP, but I only have a user account for fusion, if I open the app, it just says it can't connect as I'm not connected to the internet

**Reply by Kugelfang666** [reply]
I think your confusing some things

You can join the local WiFi AP with a pre shared key (check the sticker on the inverter) 

Once you are in you need to authenticate either as installer or owner. 

I don't remember on detail but I think DHCP could be changed even as owner

**Reply by codex70** [reply]
I managed to log in to the access point, but as a user couldn't see any way to change the DHCP settings. I suspect I need to be an installer to change those settings, but will see if I can find anything.
